### ISSUE TITLE
end time to early, cuts off the last word  

### ISSUE THREAD

**Issue by saltarob** [linto-ai/whisper-timestamped#108]
Hi,

first thank you so much for your great efforts. I really appreciate it.

I am making some tests with the large-v2 model and I noticed that most of the subtitles end to early. The start time is usually fine but the end time cuts off the last word or half the word. 

Can I address this with some changes in the settings or with using another model? I am testing English and Turkish files.

Thank you.

**Comment by josephjorgensen** [comment]
I'm experiencing the same issue. I can provide a test file privately, if that helps to debug it. It's missing the last two words ("I tried"), in my case.

**Comment by Jeronymous** [comment]
I would be interested into having files and options to reproduce that

**Comment by saltarob** [comment]
Thank you for your answer. I can provide you such files. How shall I send them to you?

**Comment by Jeronymous** [comment]
>How shall I send them to you?

You can either enclose a zip in this discussion (if you don't mind letting that public, at least for a moment).
Or you can find my email by tweaking a bit on github (Jérôme Louradour on LinkedIn if you have troubles)

**Comment by saltarob** [comment]
Thank you. Here is one of our production files. In the first link is the audio file and the subtitle file. In the second link some screen shots with the files opened in Jubler to make it visible how the last words are cut off. 

https://saltanat-test.s3-eu-west-1.amazonaws.com/public/whisper-timestamped/Audio%2BSubtitle.zip
https://saltanat-test.s3-eu-west-1.amazonaws.com/public/whisper-timestamped/JublerScreenShots.zip

Here a example screen shot:

![12th line](https://github.com/linto-ai/whisper-timestamped/assets/7444144/a9d6d8e3-baaa-407c-b53f-47f6d8a37c36)

**Comment by LaurinmyReha** [comment]
Check out this variant of whisper that was specifically designed to improve timestamps and halucinations. For english this should improve things dramatically. 

https://github.com/nyrahealth/CrisperWhisper

Feel free to also checkout the paper:

https://github.com/nyrahealth/CrisperWhisper

**Comment by saltarob** [comment]
Thank you for your suggestion. But your application seems not to work. It quits with "zsh: killed     python transcribe.py —f” without any obvious reason.

**Comment by saltarob** [comment]
With the Streamlit App I get a text in a box but no subtitles? The video player shows subtitles with single words which is completely useless. And the accuracy is very bad. And I am not interested in transcription of “UH, UH, UH”

**Comment by LaurinmyReha** [comment]
i would be curious to see this audio :) If you share it with me i would be glad to help you out.

**Comment by LaurinmyReha** [comment]
look, if you want to be like this i cant help you. Good luck with your problem!

**Comment by saltarob** [comment]
Well I wasted the whole afternoon with your app which is not working. Why did you suggest if it is not suitable for making subtitles? I don’t understand your intentions. Why should I waste your time and my time further if it is already clear that it is not suitable? And if you think it is suitable and I just use it wrongly you have the opportunity to explain it instead of being offended.

**Comment by LaurinmyReha** [comment]
Well since the model provides decent timestamps and transcription accuracy for english as proven in the paper and provides great timestamps converting these to subtitles format should be a piece of cake ( any chatbot could help you with that) so lets go to the transcription options. If you are familiar with the transformers library and Whisper all of the transcription options available in transformers are available to you. 

https://huggingface.co/nyrahealth/CrisperWhisper

Its hard to know what goes wrong with your specific audio. If it is turkish then i can not guarantee for anything performance wise since the model was not trained on turkish further. If it is english i would be glad to help you further which is hard to do without access to the audio file that you seem to have problems with.

**Comment by saltarob** [comment]
Okay fine, thank you. It’s English, but not a native speaker. With whisper-timestamped and the large-2 and large-3 model I have very good results of accuracy with that kind of recordings. I did the file already with whisper-timestamped. Where and how do you want me to provide the audio file. It’s a wav file with 100 MB.

**Comment by LaurinmyReha** [comment]
nice! Well the easiest would be probably if you could upload it to google drive and share the link with my under laurin@myreha.ai :)

**Comment by Jeronymous** [comment]
Peace and love, guys ☮️

**Comment by Jeronymous** [comment]
whisper-timespamped issues is not the place where to fight about CrisperWhisper's issues.

**Comment by saltarob** [comment]
Yes you are absolutely right. Sorry for that. On 10 Sep 2024, at 00:56, Jérôme Louradour ***@***.***> wrote:﻿
whisper-timespamped issues is not the place where to fight about CrisperWhisper's issues.
